plugins/funind/recdef.ml

```diff
@@ -1469,15 +1459,17 @@ let (com_eqn : int -> Id.t ->
 	| ConstRef c -> is_opaque_constant c
 	| _ -> anomaly ~label:"terminate_lemma" (Pp.str "not a constant.")
     in
-    let evd, env = Pfedit.get_current_context () in
-    let evd = Evd.from_ctx (Evd.evar_universe_context evd) in
+    (* let evd, env = Pfedit.get_current_context () in *)
+    let env = Global.env () in
+    (* let evd = Evd.from_ctx (Evd.evar_universe_context evd) in *)
+    let evd = Evd.from_env env in
```

Comment:
Hasn't the universe state changed here compared to before this patch? There's some leftover commented code.

Reply:
That is such a mess; even before this patch what was this thing doing? The tight context is passed now to the hook tho, so we could just pass it to `com_eqn` if we want.

Reply:
Pushed a commit clarifying a bit this code.

Reply:
> That is such a mess; even before this patch what was this thing doing? The right context is passed now to the hook tho, so we could just pass it to `com_eqn` if we want.

Honestly I don't understand what the code was doing, so maybe a safe strategy is to try to keep things unchanged, when not too costly. But I really have little insight on this. Maybe @ppedrot can guess a bit more about what we should do.